web/src/HeaderBar.tsx

```diff
@@ -0,0 +1,98 @@
+import React from "react"
+import styled from "styled-components"
+import { ReactComponent as LogoWordmarkSvg } from "./assets/svg/logo-wordmark.svg"
+import { GlobalNav } from "./GlobalNav"
+import { usePathBuilder } from "./PathBuilder"
+import { ResourceStatusSummary } from "./ResourceStatusSummary"
+import { useSnapshotAction } from "./snapshot"
+import {AnimDuration, Color, Font, FontSize, SizeUnit} from "./style-helpers"
+import { TargetType } from "./types"
+import { showUpdate } from "./UpdateDialog"
+import { Link } from "react-router-dom"
+
+const HeaderBarRoot = styled.div`
+  display: flex;
+  align-items: center;
+  padding-left: ${SizeUnit(1)};
+`
+
+const Logo = styled(LogoWordmarkSvg)`
+  justify-self: flex-start;
+  & .fillStd {
+    transition: fill ${AnimDuration.short} ease;
+    fill: ${Color.grayLightest};
+  }
+  &:hover .fillStd,
+  &.isSelected .fillStd {
+    fill: ${Color.gray7};
+  }
+  display: block;
+`
+
+const ResourceStatusSummaryContainer = styled.div`
+  justify-self: center;
+  flex-grow: 1;
+  display: flex;
+  justify-content: center;
+`
+const HeaderDivider = styled.div`
+  border-left: 1px solid ${Color.grayLighter};
+  height: ${SizeUnit(1)};
+  margin: ${SizeUnit(0.5)};
+`
+
+const AllResourcesLink = styled(Link)`
+  font-family: ${Font.monospace};
+  color: ${Color.gray7};
+  font-size: ${FontSize.small};
+  text-decoration: none;
+`
+
+type HeaderBarProps = {
+  view: Proto.webviewView
+}
+
+export default function HeaderBar(props: HeaderBarProps) {
+  let isSnapshot = usePathBuilder().isSnapshot()
+  let snapshot = useSnapshotAction()
+  let view = props.view
+  let runningBuild = view?.runningTiltBuild
+  let suggestedVersion = view?.suggestedTiltVersion
+  let resources = view?.resources || []
+  let hasK8s = resources.some((r) => {
+    let specs = r.specs ?? []
+    return specs.some((spec) => spec.type === TargetType.K8s)
+  })
+  let showMetricsButton = !!(hasK8s || view?.metricsServing?.mode)
+  let metricsServing = view?.metricsServing
+
+  let globalNavProps = {
+    isSnapshot,
+    snapshot,
+    showUpdate: showUpdate(view),
+    suggestedVersion,
+    runningBuild,
+    showMetricsButton,
+    metricsServing,
+    tiltCloudUsername: view.tiltCloudUsername ?? "",
+    tiltCloudSchemeHost: view.tiltCloudSchemeHost ?? "",
+    tiltCloudTeamID: view.tiltCloudTeamID ?? "",
+    tiltCloudTeamName: view.tiltCloudTeamName ?? "",
+  }
+
+  const pb = usePathBuilder()
+
+  return (
+    <HeaderBarRoot>
+      <Link to='/overview'>
+        <Logo width="57px" />
+      </Link>
+      <HeaderDivider />
+      <AllResourcesLink to={pb.encpath`/r/(all)/overview`} >All Resources</AllResourcesLink>
+      <ResourceStatusSummaryContainer>
```

Comment:
`ResourceStatusSummary` needs `flex-grow: 1`. Previously, we put this on `ResourceStatusSummaryRoot`, but that's kind of awkward since it prevents its parent from controlling where its child goes. (ditto for `GlobalNav` in `OverviewResourceBar`). It seems good for `HeaderBar` to control the layout of the `HeaderBar` flexbox.

I didn't see an existing pattern for this, so I took a stab and added a couple `____Container` classes to control their childrens' layout and see how it feels.

Another couple options:
* expose `style` properties on components and then do `<ResourceStatusSummary style={{flexGrow: 1}}>`
* instead of defining a `ResourceStatusSummaryContainer` const, just stick an inline `<div style={{flexGrow: 1}}>` right here. This is kind of appealing since the const isn't really doing that much for us, but it does feel kind nice to keep this a css-free zone.

This is probably a bit bikesheddy and I don't think I have a strong preference and it I don't see a reason it matters much as long as we have some kind of convention (and even then, that won't matter unless the need for this comes up more often).

Reply:
realized I could remove these elements and deal with layout from the parent's styled component, like this:
https://github.com/tilt-dev/tilt/blob/f0fefeef99e0eba00b43464e9cdc22774a940189/web/src/HeaderBar.tsx#L21-L25